meta-mender-core/recipes-mender/mender-client/mender-client_git.bb

```diff
@@ -9,5 +9,22 @@ def mender_version_of_this_recipe(d, srcpv):
         return version
 PV = "${@mender_version_of_this_recipe(d, '${SRCPV}')}"
 
+
+# If building from 'externalsrc' with client version
+# 2.1.x, 2.2.x, or 2.3.x, the systemd service file is
+# still called 'mender' with no client extension.
+def mender_client_service_name(d, srcpv):
+    version = mender_version_from_preferred_version(d, srcpv)
```

Comment:
It doesn't work. The thing is that we need the "branch" here, not the version. I don't know how to solve this.